#!/usr/bin/env python
PACKAGE = "comau_tools"
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("en_x",   bool_t,   0, "enables x", False)
gen.add("en_y",   bool_t,   0, "enables y", False)
gen.add("en_z",   bool_t,   0, "enables z", False)
gen.add("en_roll",   bool_t,   0, "enables roll", False)
gen.add("en_pitch",   bool_t,   0, "enables pitch", False)
gen.add("en_yaw",   bool_t,   0, "enables yaw", False)
gen.add("en_absolute_mode",   bool_t,   0, "enables absolute", False)
gen.add("en_relative_mode",   bool_t,   0, "enables relative", False)

gen.add("amplitude", double_t,    0, "Amplitude of sin wave (m or m/s)",  0, 0.0001, 5)
gen.add("frequency", double_t,    0, "Frequency of sin wave",  0, 0.25, 10)
gen.add("cycles",    int_t,    0, "The number of cycles of the sin", 0,  1, 20)
gen.add("start",   bool_t,   0, "When this is true the publisher starts", False)

exit(gen.generate(PACKAGE, "twist_sin_pub", "SnsTrkPlot"))
